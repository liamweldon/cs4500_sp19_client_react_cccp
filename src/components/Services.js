import React from "react";
import "./table.scss";

const Services = ({
  services,
  service,
  findAllServices,
  createService,
  deleteService,
  selectService,
  updateService,
  updateNameInForm,
  updateDescriptionInForm
}) => (
  <div>
    <h3>Services</h3>
    <table className="table">
      <tbody>
        <tr key={"header"}>
          <td>
            <div className="row">
              <div className="col-5">
                <h5>Service Name</h5>
              </div>
              <div className="col-5">
                <h5>Description</h5>
              </div>
            </div>
          </td>
        </tr>
        <tr key={"edit"}>
          <td>
            <div className="row">
              <div className="col-5">
                <input
                  onChange={e => updateNameInForm(e)}
                  value={service.serviceName}
                />
              </div>
              <div className="col-5">
                <input
                  onChange={e => updateDescriptionInForm(e)}
                  value={service.serviceDescription}
                />
              </div>
              <div className="col-1">
                <button
                  onClick={createService}
                  style={{ backgroundColor: "blue", borderColor: "blue" }}
                >
                  <i
                    className="fa fa-plus"
                    style={{ color: "white", fontSize: "50px" }}
                  />
                </button>
              </div>
              <div className="col-1">
                <button
                  onClick={updateService}
                  style={{ backgroundColor: "green", borderColor: "green" }}
                >
                  <i
                    className="fa fa-check"
                    style={{ color: "white", fontSize: "50px" }}
                  />
                </button>
              </div>
            </div>
          </td>
        </tr>
        {services.map(serv => (
          <tr className="serviceRow" key={serv.id}>
            <td>
              <div className="row">
                <div className="col-5">
                  <a href={"/admin/services/" + serv.id}>{serv.serviceName}</a>
                </div>
                <div className="col-5">
                  <a href={"/admin/services/" + serv.id}>
                    {serv.serviceDescription}
                  </a>
                </div>
                <div className="col-1">
                  <button
                    className={"delete-button" + serv.id}
                    onClick={() => deleteService(serv.id)}
                    style={{ backgroundColor: "red", borderColor: "red" }}
                  >
                    <i
                      className="fa fa-times"
                      style={{ color: "white", fontSize: "50px" }}
                    />
                  </button>
                </div>
                <div className="col-1">
                  <button
                    onClick={() => selectService(serv.id)}
                    style={{ backgroundColor: "blue", borderColor: "blue" }}
                  >
                    <i
                      className="fa fa-edit"
                      style={{ color: "white", fontSize: "50px" }}
                    />
                  </button>
                </div>
              </div>
            </td>
          </tr>
        ))}
      </tbody>
    </table>
  </div>
);

export default Services;
