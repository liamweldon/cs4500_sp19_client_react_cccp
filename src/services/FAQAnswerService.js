import {API_ROOT} from '../api-config';

export default class FAQAnswerService {
    static instance = null;
    static getInstance() {
        if (FAQAnswerService.instance === null) {
            FAQAnswerService.instance = new FAQAnswerService();
        }
        return this.instance;
    }
    findFAAById = (id) => fetch(`${API_ROOT}/api/faas/${id}`).then((response) => response.json());
    findAllFAAs = () => fetch(`${API_ROOT}/api/faas`).then((response) => response.json());
    addFAA = (newAnswer) =>
        fetch(`${API_ROOT}/api/faas/`, {
            method: 'POST',
            headers: {
                'Content-Type': 'application/json'
            },
            body: JSON.stringify(newAnswer)
        }).then((response) => response.json());

    editFAA = (id, newAnswer) =>
        fetch(`${API_ROOT}/api/faas/${id}`, {
            method: 'PUT',
            headers: {
                'Content-Type': 'application/json'
            },
            body: JSON.stringify(newAnswer)
        }).then((response) => response.json());
}