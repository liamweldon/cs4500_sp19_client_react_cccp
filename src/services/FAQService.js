import {API_ROOT} from '../api-config';

export default class FAQService {
  static instance = null;
  static getInstance() {
    if (FAQService.instance === null) {
      FAQService.instance = new FAQService();
    }
    return this.instance;
  }

  addFAQ = (newQuestion) =>
    fetch(`${API_ROOT}/api/faqs/`, {
      method: 'POST',
      headers: {
        'Content-Type': 'application/json'
      },
      body: JSON.stringify(newQuestion)
    }).then((response) => response.json());

  editFAQ = (id, newQuestion) =>
    fetch(`${API_ROOT}/api/faqs/${id}`, {
      method: 'PUT',
      headers: {
        'Content-Type': 'application/json'
      },
      body: JSON.stringify(newQuestion)
    }).then((response) => response.json());
  findFAQById = (id) => fetch(`${API_ROOT}/api/faqs/${id}`).then((response) => response.json());
  deleteFAQ = (id) => fetch(`${API_ROOT}/api/faqs/${id}`, {method: 'DELETE'}).then((response) => response.json());
  findAllFAQs = () => fetch(`${API_ROOT}/api/faqs`).then((response) => response.json());
  linkFAAtoFAQ = (faaID, faqID) => fetch(`${API_ROOT}/api/faqs/${faqID}/faas/${faaID}`).then(response => response.json());
}
