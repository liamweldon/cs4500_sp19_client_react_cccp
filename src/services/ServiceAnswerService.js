export default class ServiceAnswerService {
    static instance = null;
    static getInstance() {
        if(ServiceAnswerService.instance === null) {
            ServiceAnswerService.instance = new ServiceAnswerService()
        }
        return this.instance
    }
    findServiceAnswerById = id =>
    fetch(`http://localhost:8080/api/service-answers/${id}`)
    .then(response => response.json())
    findAllServiceAnswers = () =>
    fetch("http://localhost:8080/api/service-answers")
    .then(response => response.json())
}