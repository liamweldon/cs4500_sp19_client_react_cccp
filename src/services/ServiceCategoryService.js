export default class ServiceCategoryService {
    static instance = null;
    static getInstance() {
        if(ServiceCategoryService.instance === null) {
            ServiceCategoryService.instance = new ServiceCategoryService()
        }
        return this.instance
    }
    findServiceCategoryById = categoryId =>
        fetch(`http://localhost:8080/api/service-categories/${categoryId}`)
            .then(response => response.json())
    findAllServiceCategories = () =>
        fetch("http://localhost:8080/api/service-categories")
            .then(response => response.json())
}
