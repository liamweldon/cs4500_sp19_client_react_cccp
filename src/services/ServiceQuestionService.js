import {API_ROOT} from '../api-config';

export default class ServiceQuestionService {
    static instance = null;
    static getInstance() {
        if (ServiceQuestionService.instance === null) {
            ServiceQuestionService.instance = new ServiceQuestionService();
        }
        return this.instance;
    }
    findServiceQuestionById = (id) =>
        fetch(`${API_ROOT}/api/service-questions/${id}`).then((response) => response.json());

    findAllServiceQuestions = () => fetch(`${API_ROOT}/api/service-questions`).then((response) => response.json());

    addServiceQuestion = (newServiceQuestion) =>
    fetch(`${API_ROOT}/api/service-questions/`, {
      method: 'POST',
      headers: {
        'Content-Type': 'application/json'
      },
      body: JSON.stringify(newServiceQuestion)
    }).then((response) => response.json());
}
