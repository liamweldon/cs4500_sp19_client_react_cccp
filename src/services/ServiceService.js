import { API_ROOT } from "../api-config";

export default class ServiceService {
  static instance = null;

  static getInstance() {
    if (ServiceService.instance === null) {
      ServiceService.instance = new ServiceService();
    }
    return this.instance;
  }

  findServiceById = serviceId =>
    fetch(`${API_ROOT}/api/services/${serviceId}/`).then(response =>
      response.json());

  findAllServices = () =>
    fetch(`${API_ROOT}/api/services`).then(response => response.json());

  createService = service => {
    delete service.id;
    return fetch(`${API_ROOT}/api/services`,
    {
      method: 'post',
      body: JSON.stringify(service),
      headers: {'content-type': 'application/json'}
    })
    .then(response => response.json())
  };

  updateService = service => {
    return fetch(`${API_ROOT}/api/services/${service.id}`,
    {
      method: 'put',
      body: JSON.stringify(service),
      headers: {'content-type': 'application/json'}
    })
    .then(response => response.json())
  };

  deleteService = id => {
    return fetch(`${API_ROOT}/api/services/${id}`,
    {
      method: 'delete'
    })
  };

}
