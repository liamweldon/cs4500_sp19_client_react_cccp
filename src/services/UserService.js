import {API_ROOT} from '../api-config';

export default class UserService {
    static instance = null;

    static getInstance() {
        if (UserService.instance === null) {
            UserService.instance = new UserService();
        }
        return this.instance;
    }

    findUserById = (userId) => fetch(`${API_ROOT}/api/users/${userId}`).then((response) => response.json());
    findAllUsers = () => fetch(`${API_ROOT}/api/users`).then((response) => response.json());
    // TODO update urls
    createUser = user =>
        fetch(`${API_ROOT}/api/users`,
            {
                method: 'POST',
                body: JSON.stringify(user),
                headers: {'content-type': 'application/json'}
            })
            .then(response => response.json())

    updateUser = user =>
        fetch(`${API_ROOT}/api/users/${user.id}`,
            {
                method: 'PUT',
                body: JSON.stringify(user),
                headers: {'content-type': 'application/json'}
            })
            .then(response => response.json())

    deleteUser = user =>
        fetch(`${API_ROOT}/api/users/${user.id}`,
            {
                method: 'DELETE',
            })
}
