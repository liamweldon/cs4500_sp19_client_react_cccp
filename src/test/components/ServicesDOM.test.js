import React from 'react'
import TestRenderer from 'react-test-renderer'
import Services from '../../components/Services'
import ServiceService from '../../services/ServiceService'
import services from '../mockData/services.mock.json'
import mockService from '../services/mocks/MockServiceService'

// Provide an initial service to populate the edit fields. Replicate the one in the container
// for consistency.
let mock_initial_service = {id: 0, serviceName: "Service Name", serviceDescription: "Description"}

// Mock the service here
const serviceService = ServiceService.getInstance();
beforeEach(() => {
	global.fetch = mockService(services);
})

// DOM testing code
test('Deletion renders correctly in DOM', () => {

  // Create snapshot of initial services
  return serviceService.findAllServices().then(services => {
    let component = TestRenderer
      .create(<Services services={services}
                       service={mock_initial_service}/>);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    let testInstance = component.root;
    let rows = testInstance.findAllByProps({className : "serviceRow"});
    expect(rows.length).toBe(3);

    // Delete a service via our mocked ServiceService. In the normal case,
    // the ServiceService would modify the database, and the ServiceContainer 
    // would then update its state with the new data and re-call the Services 
    // funcional component. Here, we simulate that process by updating our 
    // mock data via the mock ServiceService and then recreating our 
    // functional component with the modified mock data.
    serviceService.deleteService(123).then(() => {
      component = TestRenderer
        .create(<Services services={services}
                        service={mock_initial_service} />);
      let tree = component.toJSON();
      expect(tree).toMatchSnapshot();

      let newTestInstance = component.root;
      let newRows = testInstance.findAllByProps({className : "serviceRow"});
      expect(newRows.length).toBe(2);
    }).catch(function(error) {});
  }).catch(function(error) {});
});
