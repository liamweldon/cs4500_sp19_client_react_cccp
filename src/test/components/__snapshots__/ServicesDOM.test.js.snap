// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Deletion renders correctly in DOM 1`] = `
<div>
  <h3>
    Services
  </h3>
  <table
    className="table"
  >
    <tbody>
      <tr>
        <td>
          <div
            className="row"
          >
            <div
              className="col-5"
            >
              <h5>
                Service Name
              </h5>
            </div>
            <div
              className="col-5"
            >
              <h5>
                Description
              </h5>
            </div>
          </div>
        </td>
      </tr>
      <tr>
        <td>
          <div
            className="row"
          >
            <div
              className="col-5"
            >
              <input
                onChange={[Function]}
                value="Service Name"
              />
            </div>
            <div
              className="col-5"
            >
              <input
                onChange={[Function]}
                value="Description"
              />
            </div>
            <div
              className="col-1"
            >
              <button
                style={
                  Object {
                    "backgroundColor": "blue",
                    "borderColor": "blue",
                  }
                }
              >
                <i
                  className="fa fa-plus"
                  style={
                    Object {
                      "color": "white",
                      "fontSize": "50px",
                    }
                  }
                />
              </button>
            </div>
            <div
              className="col-1"
            >
              <button
                style={
                  Object {
                    "backgroundColor": "green",
                    "borderColor": "green",
                  }
                }
              >
                <i
                  className="fa fa-check"
                  style={
                    Object {
                      "color": "white",
                      "fontSize": "50px",
                    }
                  }
                />
              </button>
            </div>
          </div>
        </td>
      </tr>
      <tr
        className="serviceRow"
      >
        <td>
          <div
            className="row"
          >
            <div
              className="col-5"
            >
              <a
                href="/admin/services/123"
              >
                House Cleaners
              </a>
            </div>
            <div
              className="col-5"
            >
              <a
                href="/admin/services/123"
              >
                Cleaning for your house!
              </a>
            </div>
            <div
              className="col-1"
            >
              <button
                className="delete-button123"
                onClick={[Function]}
                style={
                  Object {
                    "backgroundColor": "red",
                    "borderColor": "red",
                  }
                }
              >
                <i
                  className="fa fa-times"
                  style={
                    Object {
                      "color": "white",
                      "fontSize": "50px",
                    }
                  }
                />
              </button>
            </div>
            <div
              className="col-1"
            >
              <button
                onClick={[Function]}
                style={
                  Object {
                    "backgroundColor": "blue",
                    "borderColor": "blue",
                  }
                }
              >
                <i
                  className="fa fa-edit"
                  style={
                    Object {
                      "color": "white",
                      "fontSize": "50px",
                    }
                  }
                />
              </button>
            </div>
          </div>
        </td>
      </tr>
      <tr
        className="serviceRow"
      >
        <td>
          <div
            className="row"
          >
            <div
              className="col-5"
            >
              <a
                href="/admin/services/234"
              >
                Interior Painters
              </a>
            </div>
            <div
              className="col-5"
            >
              <a
                href="/admin/services/234"
              >
                Painting for your home!
              </a>
            </div>
            <div
              className="col-1"
            >
              <button
                className="delete-button234"
                onClick={[Function]}
                style={
                  Object {
                    "backgroundColor": "red",
                    "borderColor": "red",
                  }
                }
              >
                <i
                  className="fa fa-times"
                  style={
                    Object {
                      "color": "white",
                      "fontSize": "50px",
                    }
                  }
                />
              </button>
            </div>
            <div
              className="col-1"
            >
              <button
                onClick={[Function]}
                style={
                  Object {
                    "backgroundColor": "blue",
                    "borderColor": "blue",
                  }
                }
              >
                <i
                  className="fa fa-edit"
                  style={
                    Object {
                      "color": "white",
                      "fontSize": "50px",
                    }
                  }
                />
              </button>
            </div>
          </div>
        </td>
      </tr>
      <tr
        className="serviceRow"
      >
        <td>
          <div
            className="row"
          >
            <div
              className="col-5"
            >
              <a
                href="/admin/services/321"
              >
                Math Tutoring
              </a>
            </div>
            <div
              className="col-5"
            >
              <a
                href="/admin/services/321"
              >
                Algebra or calculus?
              </a>
            </div>
            <div
              className="col-1"
            >
              <button
                className="delete-button321"
                onClick={[Function]}
                style={
                  Object {
                    "backgroundColor": "red",
                    "borderColor": "red",
                  }
                }
              >
                <i
                  className="fa fa-times"
                  style={
                    Object {
                      "color": "white",
                      "fontSize": "50px",
                    }
                  }
                />
              </button>
            </div>
            <div
              className="col-1"
            >
              <button
                onClick={[Function]}
                style={
                  Object {
                    "backgroundColor": "blue",
                    "borderColor": "blue",
                  }
                }
              >
                <i
                  className="fa fa-edit"
                  style={
                    Object {
                      "color": "white",
                      "fontSize": "50px",
                    }
                  }
                />
              </button>
            </div>
          </div>
        </td>
      </tr>
      <tr
        className="serviceRow"
      >
        <td>
          <div
            className="row"
          >
            <div
              className="col-5"
            >
              <a
                href="/admin/services/456"
              >
                Test1
              </a>
            </div>
            <div
              className="col-5"
            >
              <a
                href="/admin/services/456"
              >
                Test1!
              </a>
            </div>
            <div
              className="col-1"
            >
              <button
                className="delete-button456"
                onClick={[Function]}
                style={
                  Object {
                    "backgroundColor": "red",
                    "borderColor": "red",
                  }
                }
              >
                <i
                  className="fa fa-times"
                  style={
                    Object {
                      "color": "white",
                      "fontSize": "50px",
                    }
                  }
                />
              </button>
            </div>
            <div
              className="col-1"
            >
              <button
                onClick={[Function]}
                style={
                  Object {
                    "backgroundColor": "blue",
                    "borderColor": "blue",
                  }
                }
              >
                <i
                  className="fa fa-edit"
                  style={
                    Object {
                      "color": "white",
                      "fontSize": "50px",
                    }
                  }
                />
              </button>
            </div>
          </div>
        </td>
      </tr>
      <tr
        className="serviceRow"
      >
        <td>
          <div
            className="row"
          >
            <div
              className="col-5"
            >
              <a
                href="/admin/services/567"
              >
                Test2
              </a>
            </div>
            <div
              className="col-5"
            >
              <a
                href="/admin/services/567"
              >
                Test2!
              </a>
            </div>
            <div
              className="col-1"
            >
              <button
                className="delete-button567"
                onClick={[Function]}
                style={
                  Object {
                    "backgroundColor": "red",
                    "borderColor": "red",
                  }
                }
              >
                <i
                  className="fa fa-times"
                  style={
                    Object {
                      "color": "white",
                      "fontSize": "50px",
                    }
                  }
                />
              </button>
            </div>
            <div
              className="col-1"
            >
              <button
                onClick={[Function]}
                style={
                  Object {
                    "backgroundColor": "blue",
                    "borderColor": "blue",
                  }
                }
              >
                <i
                  className="fa fa-edit"
                  style={
                    Object {
                      "color": "white",
                      "fontSize": "50px",
                    }
                  }
                />
              </button>
            </div>
          </div>
        </td>
      </tr>
      <tr
        className="serviceRow"
      >
        <td>
          <div
            className="row"
          >
            <div
              className="col-5"
            >
              <a
                href="/admin/services/678"
              >
                Test3
              </a>
            </div>
            <div
              className="col-5"
            >
              <a
                href="/admin/services/678"
              >
                Test3!
              </a>
            </div>
            <div
              className="col-1"
            >
              <button
                className="delete-button678"
                onClick={[Function]}
                style={
                  Object {
                    "backgroundColor": "red",
                    "borderColor": "red",
                  }
                }
              >
                <i
                  className="fa fa-times"
                  style={
                    Object {
                      "color": "white",
                      "fontSize": "50px",
                    }
                  }
                />
              </button>
            </div>
            <div
              className="col-1"
            >
              <button
                onClick={[Function]}
                style={
                  Object {
                    "backgroundColor": "blue",
                    "borderColor": "blue",
                  }
                }
              >
                <i
                  className="fa fa-edit"
                  style={
                    Object {
                      "color": "white",
                      "fontSize": "50px",
                    }
                  }
                />
              </button>
            </div>
          </div>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
